pkg/schedule/schedulers/evict_slow_store.go

```diff
@@ -102,15 +133,86 @@ func (conf *evictSlowStoreSchedulerConfig) evictStore() uint64 {
 	return conf.getStores()[0]
 }
 
+func (conf *evictSlowStoreSchedulerConfig) getPausedNetworkSlowStores() []uint64 {
+	conf.RLock()
+	defer conf.RUnlock()
+	return conf.PausedNetworkSlowStores
+}
+
+func (conf *evictSlowStoreSchedulerConfig) pauseNetworkSlowStore(storeID uint64) error {
+	conf.Lock()
+	defer conf.Unlock()
+	oldValue := conf.PausedNetworkSlowStores
+	conf.PausedNetworkSlowStores = append(conf.PausedNetworkSlowStores, storeID)
+	err := conf.save()
+	if err != nil {
+		log.Warn("failed to persist evict slow store config",
+			zap.Uint64("store-id", storeID),
+			zap.Error(err))
+		conf.PausedNetworkSlowStores = oldValue
+	}
+	return err
+}
+
+func (conf *evictSlowStoreSchedulerConfig) deleteNetworkSlowStore(storeID uint64, cluster sche.SchedulerCluster) {
+	cluster.ResumeLeaderTransfer(storeID, constant.In)
+	evictedSlowStoreStatusGauge.DeleteLabelValues(strconv.FormatUint(storeID, 10), string(networkSlowStore))
+	delete(conf.networkSlowStoreCaptureTSs, storeID)
+
+	conf.Lock()
+	defer conf.Unlock()
+
+	oldPausedNetworkSlowStores := conf.PausedNetworkSlowStores
+	conf.PausedNetworkSlowStores = slices.DeleteFunc(conf.PausedNetworkSlowStores, func(val uint64) bool {
+		return val == storeID
+	})
+	if err := conf.save(); err != nil {
+		log.Warn("failed to persist evict slow store config",
+			zap.Uint64("store-id", storeID),
+			zap.Error(err))
+		conf.PausedNetworkSlowStores = oldPausedNetworkSlowStores
+		// If failed to persist, we still need to remove the store from the networkSlowStoreCaptureTSs.
+	}
+}
+
+func (conf *evictSlowStoreSchedulerConfig) addNetworkSlowStore(storeID uint64, cluster sche.SchedulerCluster) {
+	log.Info("detected network slow store, start to pause scheduler",
+		zap.Uint64("store-id", storeID))
+	conf.networkSlowStoreCaptureTSs[storeID] = time.Now()
+	if err := cluster.PauseLeaderTransfer(storeID, constant.In); err != nil {
```

Comment:
What if the networkSlowStoreCaptureTSs has been set, but the pause leader failed?